features/FEATURE_BLE/source/generic/GenericGap.cpp

```diff
@@ -589,6 +615,107 @@ ble_error_t GenericGap::connect(
     );
 }
 
+
+ble_error_t GenericGap::connect(
+    target_peer_address_type_t peerAddressType,
+    const ble::address_t &peerAddress,
+    const ConnectionParameters &connectionParams
+) {
+    if (!connectionParams.getNumberOfEnabledPhys()) {
+        return BLE_ERROR_INVALID_PARAM;
+    }
+
+    if (!is_extended_advertising_available()) {
+        phy_set_t set(connectionParams.getPhySet());
+        if (set.count() != 1 || set.get_1m() == false) {
```

Comment:
Why '== false' instead of just '!' ?

Reply:
It seemed more readable at the time. Is it an issue ?

Reply:
It might be if we had a recommended consistent way of addressing this.

This time, it's not.